cmd/thor/utils.go

```diff
@@ -250,14 +250,26 @@ func selectGenesis(ctx *cli.Context) (*genesis.Genesis, thor.ForkConfig, error)
 	}
 }
 
-func parseGenesisFile(filePath string) (*genesis.Genesis, thor.ForkConfig, error) {
-	file, err := os.Open(filePath)
-	if err != nil {
-		return nil, thor.ForkConfig{}, errors.Wrap(err, "open genesis file")
+func parseGenesisFile(uri string) (*genesis.Genesis, thor.ForkConfig, error) {
+	var reader io.Reader
+
+	if strings.HasPrefix(uri, "http://") || strings.HasPrefix(uri, "https://") {
+		res, err := http.Get(uri) // #nosec
+		if err != nil {
+			return nil, thor.ForkConfig{}, errors.Wrap(err, "http get genesis file")
+		}
+		defer res.Body.Close()
+		reader = res.Body
+	} else {
+		file, err := os.Open(uri)
+		if err != nil {
+			return nil, thor.ForkConfig{}, errors.Wrap(err, "open genesis file")
+		}
+		defer file.Close()
+		reader = file
 	}
```

Comment:
Suggestion: found this nice `io.ReadCloser` to deal also with the file and the closing part.

```suggestion
	var reader io.ReadCloser
	var err error

	if strings.HasPrefix(uri, "http://") || strings.HasPrefix(uri, "https://") {
		res, err := http.Get(uri) // #nosec
		if err != nil {
			return nil, thor.ForkConfig{}, errors.Wrap(err, "http get genesis file")
		}
		reader = res.Body
	} else {
		reader, err = os.Open(uri)
		if err != nil {
			return nil, thor.ForkConfig{}, errors.Wrap(err, "open genesis file")
		}
	}
	defer reader.Close()
```

Reply:
Nice, I committed this